### PR TITLE
Fixed window title update & occasional sqlite error for packages

### LINKED ISSUE #549: Display scan file name in titlebar
We need do display the name of the current Scan file somewhere that is visible at all times, probably in the title bar like Excel. This is very important when you are using multiple SCWB instances.

### PR DESCRIPTION
**Pull request by OmkarPh**
Fixes #549

### CODE REVIEW

File: src/services/models/packages.ts [hunk 1]
```diff
@@ -149,6 +157,7 @@ export default function packagesModel(sequelize: Sequelize) {
         allowNull: true,
         type: DataTypes.STRING,
       },
+      license_detections: jsonDataType("license_detections"),
```
Comment (AyanSinhaMahapatra) [inline src/services/models/packages.ts:160]: Was this the source of the occasional sqlite error for packages?

File: src/mainActions.ts [hunk 2]
```diff
@@ -1,144 +1,169 @@
...
   let defaultPath;
-  if (electronOs.platform() === 'linux') {
+  if (electronOs.platform() === "linux") {
     // remove the .json (or other) extention of the path.
-    defaultPath = jsonFilePath.substring(0, jsonFilePath.lastIndexOf('.')) + '.sqlite';
+    defaultPath =
+      jsonFilePath.substring(0, jsonFilePath.lastIndexOf(".")) + ".sqlite";
   } else {
     // FIXME: this is some ugly regex used to get filename with no extension.
     // see: https://stackoverflow.com/questions/4250364/how-to-trim-a-file-extension-from-a-string-in-javascript
```
Comment (AyanSinhaMahapatra) [inline src/mainActions.ts:30]: We need to document the file license when we copy code from elsewhere, and also have a common notice for all files like we have in scancode-toolkit and elsewhere: for example see https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/cache.py#L4

File: src/mainActions.ts [hunk 3]
```diff
@@ -1,144 +1,169 @@
...
-      console.log("Sqlite file path isn't valid:", sqliteFilePath);
-      return;
-    }
-    const reply: JSON_IMPORT_REPLY_FORMAT = {
-      jsonFilePath,
-      sqliteFilePath
-    }
-    mainWindow.webContents.send(IMPORT_REPLY_CHANNEL.JSON, reply);
-  });
+  dialog
```
Comment (AyanSinhaMahapatra) [inline src/mainActions.ts:35]: We could have a default here that we can set, like a specific folder that we save the sqlite file in, to save time. This is a refinement of course, no need to do this in this PR.

File: src/mainActions.ts [hunk 4]
```diff
@@ -1,144 +1,169 @@
...
-    } else {
-      console.log("Sqlite file path isn't valid:", file, sqliteFilePath);
-      return;
-    }
-  });
+  dialog
+    .showSaveDialog(mainWindow, {
+      title: "Save as a Database File",
+      defaultPath: "fileName.sqlite",
+      filters: [{ name: "SQLite File", extensions: ["sqlite"] }],
```
Comment (AyanSinhaMahapatra) [inline src/mainActions.ts:113]: use multiple lines like above for more readable code whenever possible

**Review by AyanSinhaMahapatra — APPROVED** [review]
@OmkarPh this looks good, merging this. See also comments for your future consideration.

But we need tests running on CI: both  GitHub Actions and Azure in linux/mac/windows (And also make sure local builds are working on these platforms too) for a stable release of workbench btw.

File: src/services/models/packages.ts (hunk 1 above)
Comment (OmkarPh) [reply src/services/models/packages.ts:160]: yes

File: src/mainActions.ts (hunk 3 above)
Comment (OmkarPh) [reply src/mainActions.ts:35]: yeah makes sense, maybe settings kinda option for users to set the default path

File: src/mainActions.ts (hunk 4 above)
Comment (OmkarPh) [reply src/mainActions.ts:113]: yep, usually i let eslint do the magic :p

File: src/mainActions.ts (hunk 2 above)
Comment (OmkarPh) [reply src/mainActions.ts:30]: all stackoverflow code is licensed under [CC BY-SA](https://stackoverflow.com/help/licensing) right
so, exactly how should we show this in this file

**Comment by OmkarPh** [comment]
> @OmkarPh this looks good, merging this. See also comments for your future consideration.
> 
> But we need tests running on CI: both GitHub Actions and Azure in linux/mac/windows (And also make sure local builds are working on these platforms too) for a stable release of workbench btw.

yeah
these tests should only cover the parsers or desktop app itself too ?

File: src/mainActions.ts (hunk 2 above)
Comment (AyanSinhaMahapatra) [reply src/mainActions.ts:30]: This one specifically under https://creativecommons.org/licenses/by-sa/3.0/ yes. We should probably put a line and reference to the text, but I'm not sure of the effect of a sharealike license on the whole file/repo license. @pombredanne @DennisClark what would be recommended here?

File: src/mainActions.ts (hunk 4 above)
Comment (AyanSinhaMahapatra) [reply src/mainActions.ts:113]: Yup, that's the best way, to leave these to linters, saves time.

**Comment by AyanSinhaMahapatra** [comment]
> these tests should only cover the parsers or desktop app itself too ?

Ideally a bit of both? That's not too hard to test right? minimal unit tests and end-to-end tests to ensure most of the functions/parsers and views work? And keep adding specific cases based on bug reports (reproducing by adding a test and then fixing). I'm not familiar at all with ts/react testing so food for thought for you, what would be the way forward here

**Comment by DennisClark** [comment]
I think that simply saying that some code is licensed under https://creativecommons.org/licenses/by-sa/4.0/  is sufficient and does not have a significant impact on the overall license of the project, which is all open source with the source code freely available already, and SCWB is basically a standalone application.

File: src/mainActions.ts (hunk 2 above)
Comment (pombredanne) [reply src/mainActions.ts:30]: These regex are too small to be relevant copyright-wise IMHO... but this code is also arcane, and this is best rewritten with a cleaner and more readable way.

**Pull request merged**